### PR TITLE
[1.8.2] SwordGapOverride, Shader fix, CustomShader for Chams, AutoCrystal Extrapolation improvements

### LINKED ISSUE #152: [FEATURE] OffHand - SwordGapIgnoreHealth option
add oyvey offhand

### LINKED ISSUE #188: [FEATURE] Instant Step and AutoTrap.
- [x] Make AutoTrap place on SPacketBlockChange. Make sure it doesnt block crystal bombing. ( Bonus: Also check if HelpingBlocks for a singular Block above the head can be optimized like in Selftrap)
- [x] ~~Allow Step to get out of these instant traps by activating on BlockChange too. We step could even earlier on CPacketPlayerDigging when client side breaking the block.~~ Actually this should not be an issue, as long as NoGlitchBlocks is off and we are mining normally or Speedmine ToAir is on.

### LINKED ISSUE #190: [BUG] AutoCrystal Extrapolation STILL goes through walls
If the enemy enters the hole and
Q4IBVA6C@__TAU~%_SV{9@8
 is moving very fast, extrap will calculate another location (outside the hole)
https://youtu.be/hZevRkxU8vc

### LINKED ISSUE #191: [FEATURE] Alpha & outline setting for ShaderChams
will look like this

Shaderchams are flat and ugly asf currently, Please make evolution :pray:

### PR DESCRIPTION
**Pull request by 3arthqu4ke**
Closes #152 
Closes #191 
Closes #190
Closes #188

### CODE REVIEW

**Comment by 3arthqu4ke** [comment]
Closes #152

**Comment by 3arthqu4ke** [comment]
Closes #191

File: src/main/java/me/earth/earthhack/impl/modules/render/chams/Chams.java [hunk 1]
```diff
@@ -90,6 +90,7 @@ public class Chams extends Module
     protected final GlShader waterShader = GlShader.createShader("water");
     protected final GlShader alphaShader = GlShader.createShader("alpha");
     protected final GlShader imageShader = GlShader.createShader("image");
+    // TODO: custom shader from file?
```
Comment (mdo992) [inline src/main/java/me/earth/earthhack/impl/modules/render/chams/Chams.java:93]: yes PLEASE

**Comment by 3arthqu4ke** [comment]
Closes #190

File: src/main/java/me/earth/earthhack/impl/modules/render/chams/Chams.java (hunk 1 above)
Comment (3arthqu4ke) [reply src/main/java/me/earth/earthhack/impl/modules/render/chams/Chams.java]: yes

**Comment by 3arthqu4ke** [comment]
Fixes #152 

just testing

**Pull request merged**